ledger/babbage_test.go

```diff
@@ -83,3 +86,77 @@ func TestBabbageBlockTransactions(t *testing.T) {
+
+	// Decode the CBOR data to create a BabbageBlock
+	babbageBlock := &BabbageBlock{}
+	err := babbageBlock.UnmarshalCBOR(blockData)
+	assert.NoError(t, err)
+
+	// Ensure the block is correctly unmarshaled
+	assert.NotNil(t, babbageBlock)
+	assert.NotNil(t, babbageBlock.Header)
+	assert.NotNil(t, babbageBlock.TransactionBodies)
+	assert.NotNil(t, babbageBlock.TransactionWitnessSets)
+
+	utxoBlock := babbageBlock.Utxorpc()
+
+	// Validate the resulting utxorpc.Block
+	assert.NotNil(t, utxoBlock)
+	assert.NotNil(t, utxoBlock.Header)
+	assert.NotNil(t, utxoBlock.Body)
+	assert.Equal(t, len(babbageBlock.TransactionBodies), len(utxoBlock.Body.Tx))
+
+	// Validate the header details
+	tmpHash, _ := hex.DecodeString(babbageBlock.Hash())
+	assert.Equal(t, tmpHash, utxoBlock.Header.Hash)
+	assert.Equal(t, babbageBlock.BlockNumber(), utxoBlock.Header.Height)
+	assert.Equal(t, babbageBlock.SlotNumber(), utxoBlock.Header.Slot)
+
+	// Validate transactions
+	for i, tx := range utxoBlock.Body.Tx {
+		assert.NotNil(t, tx)
+		assert.Equal(t, len(babbageBlock.TransactionBodies[i].Inputs()), len(tx.Inputs))
+		assert.Equal(t, len(babbageBlock.TransactionBodies[i].Outputs()), len(tx.Outputs))
+	}
+}
+
+type TxOutput struct {
+	Address   []byte
+	Coin      uint64
+	DatumHash []byte
+}
+
+func TestUtxorpc_DatumOptionNil(t *testing.T) {
```

Comment:
We should keep the test names prefixed with the object type, since all tests under `ledger/` are in the same namespace. This should be something like `TestBabbageTransactionOutput_Utxorpc_DatumOptionNil`